src/main/java/appeng/client/render/model/MemoryCardBakedModel.java

```diff
@@ -0,0 +1,188 @@
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+import net.minecraftforge.common.model.TRSRTransformation;
+
+import appeng.api.implementations.items.IMemoryCard;
+import appeng.api.util.AEColor;
+import appeng.client.render.cablebus.CubeBuilder;
+import appeng.core.AELog;
+
+
+class MemoryCardBakedModel implements IBakedModel
+{
+
+	private final VertexFormat format;
+
+	private final IBakedModel baseModel;
+
+	private final TextureAtlasSprite texture;
+
+	private final long hash;
+
+	private final Cache<Long, MemoryCardBakedModel> modelCache;
+
+	private final ImmutableList<BakedQuad> generalQuads;
+
+	MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture )
+	{
+		this( format, baseModel, texture, 0, createCache() );
+	}
+
+	private MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture, long hash, Cache<Long, MemoryCardBakedModel> modelCache )
+	{
+		this.format = format;
+		this.baseModel = baseModel;
+		this.texture = texture;
+		this.hash = hash;
+		this.generalQuads = ImmutableList.copyOf( this.buildGeneralQuads() );
+		this.modelCache = modelCache;
+	}
+
+	private static Cache<Long, MemoryCardBakedModel> createCache()
+	{
+		return CacheBuilder.newBuilder()
+				.maximumSize( 100 )
+				.build();
+	}
+
+	@Override
+	public List<BakedQuad> getQuads( @Nullable IBlockState state, @Nullable EnumFacing side, long rand )
+	{
+
+		List<BakedQuad> quads = this.baseModel.getQuads( state, side, rand );
+
+		if( side != null )
+		{
+			return quads;
+		}
+
+		List<BakedQuad> result = new ArrayList<>( quads.size() + this.generalQuads.size() );
+		result.addAll( quads );
+		result.addAll( this.generalQuads );
+		return result;
+	}
+
+	private List<BakedQuad> buildGeneralQuads()
+	{
+		CubeBuilder builder = new CubeBuilder( this.format );
+
+		builder.setTexture( this.texture );
+		System.out.println( Long.toHexString( this.hash ) );
+
+		for( int x = 0; x < 4; x++ )
+		{
+			for( int y = 0; y < 2; y++ )
+			{
+				final long color = this.hash >>> ( x + y * 4 ) * 4 & 0xF;
+				final AEColor col = AEColor.values()[(int) color];
+
+				builder.setColorRGB( col.mediumVariant );
+				builder.addCube( 7 + x, 8 + y, 7.5f, 7 + x + 1, 8 + y + 1, 8.5f );
+			}
+		}
+		return builder.getOutput();
+	}
+
+	@Override
+	public boolean isAmbientOcclusion()
+	{
+		return this.baseModel.isAmbientOcclusion();
+	}
+
+	@Override
+	public boolean isGui3d()
+	{
+		return this.baseModel.isGui3d();
+	}
+
+	@Override
+	public boolean isBuiltInRenderer()
+	{
+		return this.baseModel.isBuiltInRenderer();
+	}
+
+	@Override
+	public TextureAtlasSprite getParticleTexture()
+	{
+		return this.baseModel.getParticleTexture();
+	}
+
+	@Override
+	public ItemCameraTransforms getItemCameraTransforms()
+	{
+		return this.baseModel.getItemCameraTransforms();
+	}
+
+	@Override
+	public ItemOverrideList getOverrides()
+	{
+		return new ItemOverrideList( Collections.emptyList() )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 26 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1188)